swap_meet/vendor.py

```diff
@@ -1,2 +1,219 @@
+            
+        self.inventory.append(added_item)
+        return added_item
+        
+        
+    # Wave 1- SJ
+    # 4. Created instance method named remove, which takes in one item
+    # 4a. If a matching item is found, removes the item from the inventory
+    # 4b. If no matching item is found, returns False
+
+    def remove(self, removed_item):
+        self.removed_item = removed_item
+            
+        if removed_item in self.inventory: 
+            self.inventory.remove(removed_item)
+            return removed_item
+        else:
+            return False
+        
+        
+    # Wave 2- SJ
+    # 5. Created an instance method that takes one argument, id an integer, 
+    #    id represents an Item's id number in the inventory
+    # 5a. This method returns the item from the inventory with the matching id 
+    # 5b. If no match is found, the method returns None 
+    
+    def get_by_id(self, id):
+        
+        for item in self.inventory:
+            if item.id == id:
+                return item
+            
+        return None
+    
+    
+    # Wave 3- KV
+    # 6. Created instance method named swap_items, takes 3 arguments
+    # other_vendor, my_item, their item
+    # 6a. if my item isn't in my inventory, OR if their item isn't in their inventory,
+    # 6b. the return False
+    # 6c. Otherwise, remove the item from my inventory
+    # 6d. Add that item to their inventory 
+    # 6e. Likewise, if the other vendors item is in their inventory, remove that item
+    # 6f. Add that item to my inventory and retrun True
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+
+        other_vendor.inventory.append(my_item)
+
+        other_vendor.inventory.remove(their_item)
+        
+        self.inventory.append(their_item)
+    
+        return True
+        
+
+# Wave 4 - SJ
+# 7. Created an instance method swap_first_item takes one argument, other_vendor
+#    considers the first item in the instance's / self.inventory, 
+#    and the first item in the friend's/ other_vendor.inventory
+# 7a. if the inventory for either are 0, return False
+# 7b. otherwise add that first item from their inventory to mine
+# 7c. then remove that first item from their inventory 
+# 7d. likewise, add the first item from my inventory to their inventory
+# 7e. then remove my first item from my inventory
+# 7f. return True
+
+    def swap_first_item(self, other_vendor):
+        
+        if len(self.inventory) == 0  or len(other_vendor.inventory)== 0:
+            return False
+
+        self.inventory.append(other_vendor.inventory[0])
+        
+        other_vendor.inventory.remove(other_vendor.inventory[0])
+
+        other_vendor.inventory.append(self.inventory[0])
+
+        self.inventory.remove(self.inventory[0])
```

Comment:
We could condense this code by using the method `swap_items` that we wrote above this method here. Remember, we want to keep our code **DRY**.

Reply:
Good point - totally forgot about that function...